src/main/java/org/primefaces/extensions/component/fuzzysearch/FuzzySearchRenderer.java

```diff
@@ -0,0 +1,159 @@
+        }
+
+        FuzzySearch fuzzySearch = (FuzzySearch) component;
+
+        encodeMarkup(context, fuzzySearch);
+        encodeScript(context, fuzzySearch);
+    }
+
+    private void encodeScript(FacesContext context, FuzzySearch fuzzySearch) throws IOException {
+        WidgetBuilder wb = getWidgetBuilder(context);
+
+        // TODO is it possible to make improvement here to have @FuzzySearchKey annotated class?
+        List list = (List) fuzzySearch.getValue();
+        Object o = list.get(0);
+        Class<? extends Object> clazz = o.getClass();
+        Field[] fields = clazz.getDeclaredFields();
+        List<String> keys = Arrays.asList(fields).stream()
+                .filter(f -> f.getAnnotation(FuzzySearchKey.class) != null)
+                .map(Field::getName)
+                .collect(Collectors.toList());
+        if (keys.isEmpty()) {
+            throw new FacesException("No @FuzzyKey annotation was detected on your class " + clazz + "."
+                                     + " Please annotate at least one field in the class as @FuzzySearchKey.");
+        }
+
+        Gson gson = new Gson();
+        String jsonKeys = gson.toJson(keys);
+        String jsonValue = gson.toJson(fuzzySearch.getValue());
+
+        wb.init(FuzzySearch.class.getSimpleName(), fuzzySearch.resolveWidgetVar(), fuzzySearch.getClientId(context))
+                .attr("keys", jsonKeys)
+                .attr("value", jsonValue);
+
+        if (fuzzySearch.getOnSelect() != null) {
+            // Define a callback function when the item is selected
+            wb.callback("onSelect", "function(e)", fuzzySearch.getOnSelect());
+        }
+
+        encodeClientBehaviors(context, fuzzySearch);
+
+        wb.finish();
+    }
+
+    protected void encodeMarkup(FacesContext context, FuzzySearch fuzzySearch) throws IOException {
+        final ResponseWriter writer = context.getResponseWriter();
+
+        String clientId = fuzzySearch.getClientId(context);
+        String style = fuzzySearch.getStyle();
+        String styleClass = fuzzySearch.getStyleClass();
+        styleClass = (styleClass == null) ? FuzzySearch.CONTAINER_CLASS : FuzzySearch.CONTAINER_CLASS + " " + styleClass;
+        String resultStyle = fuzzySearch.getResultStyle();
+        String resultStyleClass = fuzzySearch.getResultStyleClass();
+        resultStyleClass = (resultStyleClass == null) ? FuzzySearch.ITEM_CLASS : FuzzySearch.ITEM_CLASS + " " + resultStyleClass;
+
+        writer.startElement("div", fuzzySearch);
+        writer.writeAttribute("id", clientId, null);
+        writer.writeAttribute("class", styleClass, "styleClass");
+        if (style != null) {
+            writer.writeAttribute("style", style, "style");
+        }
+        writer.writeAttribute("data-result-style", fuzzySearch.getResultStyle(), null);
+        writer.writeAttribute("data-result-style-class", fuzzySearch.getResultStyleClass(), null);
```

Comment:
I think these would be better as WidgetBuilder attributes like `wb.attr('resultStyle', fuzzySearch.getResultStyle())` so they will be on the `this.cfg` object in the Javascript side.